client/web/src/repo/utils.tsx

```diff
@@ -45,3 +47,37 @@ export const stringToCodeHostType = (codeHostType: string): CodeHostType => {
         }
     }
 }
+
+export interface FileInfo {
+    extension: FileExtension
+    isTest: boolean
+}
+
+export const getFileInfo = (file: string, isDirectory: boolean): FileInfo => {
+    const fileInfo = { extension: '' as FileExtension, isTest: false }
+    fileInfo.isTest = isDirectory ? false : containsTest(file)
+
+    if (isDirectory) {
+        return fileInfo
+    }
+
+    const f = file.split('.')
+    // Last item in 'f' is file extension string
+    fileInfo.extension = f.pop() as FileExtension
```

Comment:
Proposing this change here

```suggestion
export const getFileInfo = (file: string, isDirectory: boolean): FileInfo => {
    const isTest = isDirectory ? false : containsTest(file)

    if (isDirectory) {
        return { isTest, extension: undefined }
    }

    const f = file.split('.')
    // Last item in 'f' is file extension string
    const extension = converTotFileExtensionEnum(f.pop())
    return { extension, isTest }
    
```

Reply:
Marking this as resolved even though I'm not taking the suggestion verbatim. However, I am going to change this so that we're not casting an empty string as an enum that doesn't exist.